frontend/src/components/GameForm/JoinGameForm.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <GameForm>
+    <template v-if="!game">
+      Loading ...
+    </template>
+
+    <template v-else>
+      <b-form @submit.prevent="onSubmit">
+        <b-form-group label="Your Name" label-for="nickname">
+          <b-form-input
+            id="nickname"
+            v-model="player.nickname"
+            type="text"
+            placeholder="Zaphord"
+            required
+            minlength="3"
+          ></b-form-input>
+        </b-form-group>
+
+        <b-form-group v-if="game.hasPassword" label="Password" label-for="password">
+          <b-form-input id="password" v-model="password" type="password" required autocomplete="off"></b-form-input>
+        </b-form-group>
+
+        <b-button type="submit" variant="secondary" size="sm" class="mt-3">
+          Join game
+        </b-button>
+      </b-form>
+    </template>
+  </GameForm>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+import GameForm from './GameForm.vue';
+import { Game, CreatePlayer, MessageGetGame, MessageJoinGame, UUID } from '../../../types';
+import axios from '../../helpers/api';
+import store from '../../store';
+
+export default Vue.extend({
+  name: 'JoinGameForm',
+  components: {
+    GameForm,
+  },
+  props: {
+    gameId: {
+      type: String,
+      required: false,
+      default: '',
+    },
+  },
+  data() {
+    return {
+      game: undefined as Game | undefined,
+      password: '',
+      player: {
+        nickname: '',
+      } as CreatePlayer,
+      errors: [] as unknown[],
+    };
+  },
+  computed: {},
+  mounted() {
+    this.fetchGame();
+  },
+  methods: {
+    async fetchGame(): Promise<void> {
+      const response = await axios.get<MessageGetGame>(`/games/${this.gameId}`);
+      this.game = response.data.game;
+    },
+    onSubmit(): void {
+      const { player, password } = this;
+      const data: MessageJoinGame = { ...player, password: password || undefined };
+      store.dispatch
+        .joinGame({
+          id: this.gameId as UUID,
+          data,
+        })
+        .then(() => {
+          // reload
+          this.$router.go(0);
```

Comment:
This seems to be a bit ugly to me, is there a good way to trigger the parent component?

Reply:
Proposal: in `AccessGame` render form if no GameState exists, `InGame` otherwise.